docs/src/pages/block.js

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useCallback } from "react";
+import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
+import Layout from "@theme/Layout";
+import ExecutionEnvironment from "@docusaurus/ExecutionEnvironment";
+import CodeBlock from "@theme/CodeBlock";
+import useBaseUrl from "@docusaurus/useBaseUrl";
+import BrowserOnly from "@docusaurus/BrowserOnly";
+import Link from "@docusaurus/Link";
+
+function getQueryVariable(variable) {
+    const query = window.location.search.substring(1);
+    const vars = query.split("&");
+    for (let i = 0; i < vars.length; i++) {
+        const pair = vars[i].split("=");
+        if (decodeURIComponent(pair[0]) == variable) {
+            return decodeURIComponent(pair[1]);
+        }
+    }
+}
```

Comment:
We can use [URL.searchParams](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams) instead, it does URI decoding for us too AIUI.

```js
function getQueryVariable(variable) {
  const params = (new URL(document.location)).searchParams;
  return params.get(variable); // returns null if not found.
}
```

Reply:
👍